src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -427,6 +440,20 @@ private String getMapping() throws IOException {
             return Resources.toString(url, Charsets.UTF_8);
         }
 
+        @Override
+        public boolean isModelBlockedForDelete(String modelId) {
```

Comment:
check modelId for null or empty

Reply:
IMO we don't need to validate modelId for null, empty or blank string because it checks if that value is present in a HashSet and HashSet doesn't throw any exception and simply returns false